lms/static/js/instructor_dashboard/membership.js

```diff
@@ -603,7 +603,7 @@ such that the value can be defined later than this assignment (file load order).
             this.$request_response_error = this.$container.find('.request-response-error');
             this.$enrollment_button.click(function(event) {
                 var sendData;
-                if (!batchEnroll.$reason_field.val()) {
+                if (batchEnroll.$reason_field.length && !batchEnroll.$reason_field.val()) {
```

Comment:
For an empty string, wouldn't `batchEnroll.$reason_field.length` == 0 causing the client validation to fail?
I'm not sure what situation would result is a non-zero length and a still have `val()==false`.

Reply:
I am not the expert but `$reason_field` has an attribute of length by default which is different than string or array length function because it is a reference to an element (reason textarea). 

So basically when we do `batchEnroll.$reason_field.length` this will have only two values 1 or 0 depending upon whether that element/field ($reason_field) exists on our webpage or not. 

And `!batchEnroll.$reason_field.val()` will return the string value of reason field. So when the reason field will exist but it will have an empty value. Then our first condition will be true but `val()==false`. I hope, it helps and this check already exists in other places too. I am attaching the screenshot if you don't want to test it.

<img width="659" alt="Screenshot 2019-12-11 at 11 22 04 AM" src="https://user-images.githubusercontent.com/8234192/70597383-05284f80-1c0a-11ea-865f-ed32f415335c.png">

Reply:
Thanks for the explanation! The had assumed `.length` was just the value length, but that's not the battle we're fighting today. Seems good to me then 👍